sdk/python/kfp/components/placeholders.py

```diff
@@ -444,6 +446,7 @@ def maybe_convert_placeholder_string_to_placeholder(
 
     # order matters here!
     from_string_placeholders = [
+        ExecutorInputPlaceholder,
```

Comment:
Correct me if I'm wrong this list is used in parsing from the final serialized strings into in-memory data structure. For executor input placeholder, it's parsing from `{{$}}` rather than `{executorInput: nulll}`, am I right?
If that's the case, we can probably skip this one, as the placeholder doesn't hold any reference to inputs/outputs like others do. So loading the string `{{$}}` into the placeholder class has no meaningful operation afterwards.

Reply:
That sounds correct to me. And yes, I think this would not parse `{executorInput: null}`, which it probably ought to if we keep this approach.

I see two benefits to keeping the `ExecutorInputPlaceholder` at all:
1. Placeholder handling consistency (not unambiguously a benefit -- this also adds some complexity due to semi-superfluous parsing)
2. It can help us convert legacy `{executorInput: null}` to the preferred `{{$}}` when someone loads an old component and recompiles it

Do you think this warrants keeping the `ExecutorInputPlaceholder`? If you think `ExecutorInputPlaceholder` is worth keeping, I will modify the code to also parse `{executorInput: null}`. WDYT?

Reply:
SGTM to keep `ExecutorInputPlaceholder` here for consistency. 
`{executorInput: null}` (w/o quotes, thus treated as a dictionary in YAML syntax) will be processed elsewhere--I recall seeing some code handling `if firstKey == 'executorInput'` where we handle dictionaries like `{inputValue: ...}` etc. 
We don't need to parse a string literal `"{executorInput: null}"` or `"{inputValue: ...}:"` to extract the placeholder structure from it, only need to parse strings like `"{{$}}"` (optionally for consistency) and `"{$.inputs.parameters[...]}"`, etc.
So I think your change looks good, nothing needs to be changed.

Reply:
Sounds good to me! Thanks, Chen.